xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_dll.h

```diff
@@ -630,6 +630,11 @@ extern "C"
   void OnPowerSavingDeactivated();
 
   /*!
+   * Get stream times. Intermediate, will be moved to inputstream
+   */
+  bool GetStreamTimes(PVR_STREAM_TIMES &times);
```

Comment:
Two points here:
1) May I suggest to change the return value to PVR_ERROR. Despite from being consistent with existing API design, this gives you the chance to distinguish between some error occured due to some reason versus "not implemented". The latter could help to implement migration: first, call GetStreamTimes and if it returns NOT_IMPLEMENTED, call the deprectated counterpart functions. As an alternative, you could introduce an addon capability. This has the benefit that there is no need for the "call the new function, check for not implemented, call old functions in this case" cycle whenever the times are needed. Could make sense if the times are requested very often.

2) As this is a C-API you must not use C++ language constructs (here: pass by reference). I know we do this at other places and Alwin is fixing this for the reworked APIs, but for new functions we should do it right in the first place. Use a pointer to implement the out param, please (PVR_STREAM_TIMES *times).